documentation/CAMARA-Security-Interoperability.md

```diff
@@ -255,8 +255,9 @@ This document does not mandate a particular PPID algorithm to be used.
 
 ## Client Authentication
 
-This CAMARA document allows **one** client authentication method, `private_key_jwt`, as defined in OIDC
-[OIDC Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication)
+The API consumer MUST authenticate with the authorisation server using `private_key_jwt`, as specified in [OIDC Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication). In addition to the mandated claims, the signed JWT MUST also include the `iat` (issued at) claim.
+
+The difference between the `exp` (expires at) and `iat` (issued at) claims MUST be no more than 300 seconds. JWTs with a longer lifetime SHALL be rejected by the authorisation server using the `invalid_client` error code as specified in [IETF RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2) for the `/token` endpoint, and as specified in [OIDC CIBA Core](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.13) for the `/bc-authorize` endpoint.
```

Comment:
To avoid issues with existing implementations that strictly follow the specifications, we would prefer it to be a recommendation.

```suggestion
The API consumer MUST authenticate with the authorisation server using `private_key_jwt`, as specified in [OIDC Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication). In addition to the mandated claims, the signed JWT SHOULD also include the `iat` (issued at) claim.
 
The difference between the exp (expires at) and iat (or the authorization server's current time if absent) claims SHOULD be no more than 300 seconds. JWTs with a longer lifetime MAY be rejected by the authorisation server using the `invalid_client` error code as specified in [IETF RFC 6749](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2) for the `/token` endpoint, and as specified in [OIDC CIBA Core](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.13) for the `/bc-authorize` endpoint.
```

Reply:
The request for a hard limit on token lifetime came from @MarkCornall, who was concerned that an inconsistent API consumer experience would frustrate API consumers. There was no pushback on this during the last meeting.

The proposed text doesn't really change anything from vanilla OIDC - tokens can have any lifetime the API consumer chooses, but authorisation servers may reject tokens they consider to have an excessive lifetime without publishing in advance what this limit is.

I had understood that this document was meant to be an interoperability profile, but it is turning into a series of suggestions and recommendations which does not guarantee interoperability at all.

Reply:
> The request for a hard limit on token lifetime came from @MarkCornall, who was concerned that an inconsistent API consumer experience would frustrate API consumers. There was no pushback on this during the last meeting.
> 
> The proposed text doesn't really change anything from vanilla OIDC - tokens can have any lifetime the API consumer chooses, but authorisation servers may reject tokens they consider to have an excessive lifetime without publishing in advance what this limit is.
> 
> I had understood that this document was meant to be an interoperability profile, but it is turning into a series of suggestions and recommendations which does not guarantee interoperability at all.

To take the last paragraph. This needs to be an interoperability profile. We should take bold steps to build interoperability not suggestions.

Reply:
Text proposal updated as discussed below